diff -rcN src/gcc-4.8.1/config.sub src/gcc/config.sub
*** src/gcc-4.8.1/config.sub	2013-01-11 14:02:58.000000000 +0100
--- src/gcc/config.sub	2013-08-02 16:32:29.534835590 +0200
***************
*** 1358,1364 ****
  	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
  	      | -sym* | -kopensolaris* | -plan9* \
  	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
! 	      | -aos* | -aros* \
  	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
  	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
  	      | -hiux* | -386bsd* | -knetbsd* | -mirbsd* | -netbsd* \
--- 1358,1364 ----
  	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
  	      | -sym* | -kopensolaris* | -plan9* \
  	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
! 	      | -aos* | -aros* | -nulnova* \
  	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
  	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
  	      | -hiux* | -386bsd* | -knetbsd* | -mirbsd* | -netbsd* \
diff -rcN src/gcc-4.8.1/gcc/config/nulnova.h src/gcc/gcc/config/nulnova.h
*** src/gcc-4.8.1/gcc/config/nulnova.h	1970-01-01 01:00:00.000000000 +0100
--- src/gcc/gcc/config/nulnova.h	2013-08-02 18:05:39.946962929 +0200
***************
*** 0 ****
--- 1,31 ----
+   #undef   STARTFILE_SPEC
+   #define STARTFILE_SPEC "%{shared|pie:crt0S.o%s;:crt0.o%s} %{shared:crt1S.o%s;:crt1.o%s} %{shared|pie:crtbeginS.o%s;:crtbegin.o%s}"
+   
+   #undef   ENDFILE_SPEC
+   #define ENDFILE_SPEC "%{shared|pie:crtendS.o%s;:crtend.o%s} %{shared:crtnS.o%s;:crtn.o%s}"
+   
+   #undef LINK_SPEC
+   #define LINK_SPEC ""
+   
+   #undef TARGET_OS_CPP_BUILTINS
+   #define TARGET_OS_CPP_BUILTINS()      \
+     do {                                \
+       /* to force a strcmp for typeinfo-compares */ \
+       builtin_define ("__GXX_MERGED_TYPEINFO_NAMES=0"); \
+       builtin_define ("__GXX_TYPEINFO_EQUALITY_INLINE=0"); \
+       builtin_define ("__nova__");				\
+       builtin_define ("__NOVA__");				\
+       builtin_define_std ("nova");      \
+       builtin_assert ("system=nova");   \
+     } while(0);
+   
+   #undef LINK_GCC_C_SEQUENCE_SPEC
+   #define LINK_GCC_C_SEQUENCE_SPEC \
+     "%{static:--start-group} %G %L %{static:--end-group}%{!static:%G}"
+   
+   /* Use --as-needed -lgcc_s for eh support.  */
+   #ifdef HAVE_LD_AS_NEEDED
+   #define USE_LD_AS_NEEDED 1
+   #endif
+   
+   #define NO_IMPLICIT_EXTERN_C
diff -rcN src/gcc-4.8.1/gcc/config/t-nulnova src/gcc/gcc/config/t-nulnova
*** src/gcc-4.8.1/gcc/config/t-nulnova	1970-01-01 01:00:00.000000000 +0100
--- src/gcc/gcc/config/t-nulnova	2013-08-02 16:41:36.818848056 +0200
***************
*** 0 ****
--- 1,36 ----
+ EXTRA_PARTS=crtbegin.o crtbeginS.o crtend.o crtendS.o
+ 
+ # Build a shared libgcc library for ELF without symbol versioning
+ # with the GNU linker.
+ 
+ SHLIB_EXT = .so
+ SHLIB_SOLINK = @shlib_base_name@.so
+ SHLIB_SOVERSION = 1
+ SHLIB_SONAME = @shlib_base_name@.so.$(SHLIB_SOVERSION)
+ SHLIB_OBJS = @shlib_objs@
+ SHLIB_DIR = @multilib_dir@
+ SHLIB_SLIBDIR_QUAL = @shlib_slibdir_qual@
+ SHLIB_LC = -lc
+ SHLIB_MAKE_SOLINK = $(LN_S) $(SHLIB_SONAME) $(SHLIB_DIR)/$(SHLIB_SOLINK)
+ SHLIB_INSTALL_SOLINK = $(LN_S) $(SHLIB_SONAME) \
+ 	$$(DESTDIR)$$(slibdir)$(SHLIB_SLIBDIR_QUAL)/$(SHLIB_SOLINK)
+ 
+ SHLIB_LINK = $(GCC_FOR_TARGET) $(LIBGCC2_CFLAGS) -shared -Wl,-shared -nodefaultlibs \
+ 	-Wl,--soname=$(SHLIB_SONAME) \
+ 	-o $(SHLIB_DIR)/$(SHLIB_SONAME).tmp @multilib_flags@ \
+ 	$(SHLIB_OBJS) $(SHLIB_LC) && \
+ 	rm -f $(SHLIB_DIR)/$(SHLIB_SOLINK) && \
+ 	if [ -f $(SHLIB_DIR)/$(SHLIB_SONAME) ]; then \
+ 	  mv -f $(SHLIB_DIR)/$(SHLIB_SONAME) \
+ 		$(SHLIB_DIR)/$(SHLIB_SONAME).backup; \
+ 	else true; fi && \
+ 	mv $(SHLIB_DIR)/$(SHLIB_SONAME).tmp $(SHLIB_DIR)/$(SHLIB_SONAME) && \
+ 	$(SHLIB_MAKE_SOLINK)
+ # $(slibdir) double quoted to protect it from expansion while building
+ # libgcc.mk.  We want this delayed until actual install time.
+ SHLIB_INSTALL = \
+ 	$$(mkinstalldirs) $$(DESTDIR)$$(slibdir)$(SHLIB_SLIBDIR_QUAL); \
+ 	$(INSTALL_DATA) $(SHLIB_DIR)/$(SHLIB_SONAME) \
+ 	  $$(DESTDIR)$$(slibdir)$(SHLIB_SLIBDIR_QUAL)/$(SHLIB_SONAME); \
+ 	rm -f $$(DESTDIR)$$(slibdir)$(SHLIB_SLIBDIR_QUAL)/$(SHLIB_SOLINK); \
+ 	$(SHLIB_INSTALL_SOLINK)
diff -rcN src/gcc-4.8.1/gcc/config.gcc src/gcc/gcc/config.gcc
*** src/gcc-4.8.1/gcc/config.gcc	2013-03-13 10:47:41.000000000 +0100
--- src/gcc/gcc/config.gcc	2013-08-02 17:05:11.802880287 +0200
***************
*** 813,818 ****
--- 813,826 ----
  esac
  
  case ${target} in
+ i[3-7]86-*-nulnova*)
+ 	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h i386/i386elf.h nulnova.h"
+ 	tmake_file="t-nulnova i386/t-crtstuff t-svr4"
+ 	gas=yes
+ 	gnu_ld=yes
+ 	default_use_cxa_atexit=yes
+  	use_fixproto=yes
+  	;;
  aarch64*-*-elf)
  	tm_file="${tm_file} dbxelf.h elfos.h newlib-stdint.h"
  	tm_file="${tm_file} aarch64/aarch64-elf.h aarch64/aarch64-elf-raw.h"
diff -rcN src/gcc-4.8.1/libgcc/config.host src/gcc/libgcc/config.host
*** src/gcc-4.8.1/libgcc/config.host	2013-02-04 20:06:20.000000000 +0100
--- src/gcc/libgcc/config.host	2013-08-02 16:47:30.070856102 +0200
***************
*** 280,285 ****
--- 280,288 ----
  esac
  
  case ${host} in
+ i[3-7]86-*-nulnova*)
+ 	extra_parts="crtbegin.o crtbeginS.o crtend.o crtendS.o"
+ 	;;
  aarch64*-*-elf)
  	extra_parts="$extra_parts crtbegin.o crtend.o crti.o crtn.o"
  	tmake_file="${tmake_file} ${cpu_type}/t-aarch64"
diff -rcN src/gcc-4.8.1/libgcc/gthr-posix.h src/gcc/libgcc/gthr-posix.h
*** src/gcc-4.8.1/libgcc/gthr-posix.h	2013-02-04 20:06:20.000000000 +0100
--- src/gcc/libgcc/gthr-posix.h	2013-08-02 17:12:02.962889652 +0200
***************
*** 33,38 ****
--- 33,39 ----
  #define __GTHREADS_CXX0X 1
  
  #include <pthread.h>
+ #include <sched.h>
  
  #if ((defined(_LIBOBJC) || defined(_LIBOBJC_WEAK)) \
       || !defined(_GTHREAD_USE_MUTEX_TIMEDLOCK))
